components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/gatewayartifactsynchronizer/DBRetriever.java

```diff
@@ -143,9 +148,26 @@ public String retrieveArtifact(String APIId, String gatewayLabel, String gateway
                 throw new ArtifactSynchronizerException("HTTP response is empty");
             }
             Map <String, String> apiAttribute = new HashMap<>();
+
             JSONObject artifactObject = new JSONObject(responseString);
-            String apiId = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
-            String label = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.LABEL);
+            String label = null;
+            String apiId = null;
+            try {
+                apiId = (String)artifactObject.get(APIConstants.GatewayArtifactSynchronizer.API_ID);
+                String labelsStr = artifactObject.get(APIConstants.GatewayArtifactSynchronizer.LABELS).toString();
```

Comment:
JSONArrray

Reply:
Converting the JSONArray to set to get the intercept of two sets.